### PR TITLE
Add test template example for helm create

### LINKED ISSUE #3559: [feature request] helm create - test template
`helm create` should create `test` template example

### PR DESCRIPTION
**Pull request by hickeyma**
Test template example for `helm create.` 

Closes #3559

### CODE REVIEW

**Comment by thomastaylor312** [comment]
cc @michelleN as you may be interested in adding a test like this for `helm create`

File: cmd/helm/create_test.go [hunk 1]
```diff
@@ -143,8 +143,8 @@ func TestCreateStarterCmd(t *testing.T) {
 		t.Errorf("Wrong API version: %q", c.Metadata.ApiVersion)
 	}
 
-	if l := len(c.Templates); l != 6 {
-		t.Errorf("Expected 5 templates, got %d", l)
+	if l := len(c.Templates); l != 7 {
```
Comment (henrynash) [inline cmd/helm/create_test.go]: Although I agree that this change keeps the spirit of the original code, might it be better to define the number of expected templates as a constant and reference to the constant+1 in the test?

**Review by henrynash — COMMENTED** [review]
minor suggestion regarding number of templates referenced in test

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/chartutil/create.go [hunk 2]
```diff
@@ -290,6 +292,26 @@ Create chart name and version as used by the chart label.
...
 
+const defaultServiceTest = `apiVersion: v1
+kind: Pod
+metadata:
+  name: "{{ template "<CHARTNAME>.fullname" . }}-service-test"
+  labels:
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app: {{ template "<CHARTNAME>.name" . }}
```
Comment (unguiculus) [inline pkg/chartutil/create.go]: Labels should match those of other resources and `include` should be preferred over `template`:

```yaml
app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
helm.sh/chart: {{ include "<CHARTNAME>.chart" . }}
app.kubernetes.io/instance: {{ .Release.Name }}
app.kubernetes.io/managed-by: {{ .Release.Service }}
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: cmd/helm/create_test.go (hunk 1 above)
Comment (bacongobbler) [reply cmd/helm/create_test.go]: Generally speaking if it's not a core part of the PR (in this case, if it does not add a test template to `helm create`), we ask contributors to file a separate PR. It's a good suggestion, but I wouldn't say it's something that needs to be done to accomplish what this PR is trying to solve.

However, if you feel like submitting a new PR to improve the tests, that would be much appreciated @henrynash.

Comment (henrynash) [reply cmd/helm/create_test.go]: Agree entirely with this approach. I'll submit a follow-on PR.

Comment (hickeyma) [reply cmd/helm/create_test.go]: Ok, thanks for feedback @bacongobbler. @henrynash going to leave it as is, as its not related to the PR and not an issue either.

File: pkg/chartutil/create.go [hunk 3]
```diff
@@ -44,8 +44,10 @@ const (
 	ServiceName = "service.yaml"
 	// NotesName is the name of the example NOTES.txt file.
 	NotesName = "NOTES.txt"
-	// HelpersName is the name of the example NOTES.txt file.
+	// HelpersName is the name of the example helpers file.
 	HelpersName = "_helpers.tpl"
+	// ServiceTestName is the name of the example test service file.
+	ServiceTestName = "service-test.yaml"
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply pkg/chartutil/create.go]: Sure, done.

File: pkg/chartutil/create.go [hunk 4]
```diff
@@ -290,6 +292,26 @@ Create chart name and version as used by the chart label.
...
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    app: {{ template "<CHARTNAME>.name" . }}
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: curl
+      image: radial/busyboxplus:curl
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply pkg/chartutil/create.go]: Agree, changed to busybox `wget`.

File: pkg/chartutil/create.go [hunk 5]
```diff
@@ -399,6 +421,11 @@ func Create(chartfile *chart.Metadata, dir string) (string, error) {
 			path:    filepath.Join(cdir, TemplatesDir, HelpersName),
 			content: Transform(defaultHelpers, "<CHARTNAME>", chartfile.Name),
 		},
+		{
+			// service-test.yaml
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply pkg/chartutil/create.go]: Done.

File: pkg/chartutil/create.go [hunk 6]
```diff
@@ -399,6 +421,11 @@ func Create(chartfile *chart.Metadata, dir string) (string, error) {
...
+			path:    filepath.Join(cdir, TemplatesDir, ServiceTestName),
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply pkg/chartutil/create.go]: Yep, done.

File: pkg/chartutil/create.go (hunk 2 above)
Comment (hickeyma) [reply pkg/chartutil/create.go]: Good catch. Updated.

**Comment by hickeyma** [comment]
Thanks @henrynash @bacongobbler @unguiculus for review comments.

Updates done and ready for review again.

**Comment by bacongobbler** [comment]
Mind taking a look at the failing tests? Try rebasing first and see if that fixes a few of the tests.

**Comment by bacongobbler** [comment]
thanks. I'll give this a spin later this week. Thanks!

**Comment by hickeyma** [comment]
@bacongobbler @unguiculus Ready for review again when you get a chance.

**Review by bacongobbler — APPROVED** [review]
works as tested.

```
$ helm create foo
$ helm install ./foo
$ helm test $(helm last)
RUNNING: veering-ladybug-foo-test-connection
PASSED: veering-ladybug-foo-test-connection
```

**Comment by hickeyma** [comment]
Thanks @bacongobbler

**Comment by hickeyma** [comment]
ported in #6490

**Pull request merged**